### ISSUE TITLE
Near gps location filter

### ISSUE THREAD

**Issue by stuk88** [balderdashy/waterline-criteria#23]
Hi!
Im suggesting adding a gps proximity filtering option.
the criteria could look like that:
    where: {location: {near: gps_here}}, limit:3}

the sql query to make this happen:
```mysql
SELECT *
  FROM (
 SELECT 
        u.latitude, u.longitude,
        p.radius,
        p.distance_unit
                 * DEGREES(ACOS(COS(RADIANS(p.latpoint))
                 * COS(RADIANS(u.latitude))
                 * COS(RADIANS(p.longpoint - u.longitude))
                 + SIN(RADIANS(p.latpoint))
                 * SIN(RADIANS(u.latitude)))) AS distance
  FROM user AS u
  JOIN (   /* these are the query parameters */
        SELECT  42.81  AS latpoint,  -70.81 AS longpoint,
                50.0 AS radius,      111.045 AS distance_unit
    ) AS p ON 1=1
  WHERE u.latitude
     BETWEEN p.latpoint  - (p.radius / p.distance_unit)
         AND p.latpoint  + (p.radius / p.distance_unit)
    AND u.longitude
     BETWEEN p.longpoint - (p.radius / (p.distance_unit * COS(RADIANS(p.latpoint))))
         AND p.longpoint + (p.radius / (p.distance_unit * COS(RADIANS(p.latpoint))))
 ) AS d
 WHERE distance <= radius
 ORDER BY distance
 LIMIT 15
```

http://www.plumislandmedia.net/mysql/haversine-mysql-nearest-loc/

**Comment by mikermcneil** [comment]
Thanks for posting, @stuk88.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/waterline-criteria/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
